addon/mixins/admin-adapter-mixin.js

```diff
@@ -27,7 +37,53 @@ export default Mixin.create(RESTAdapterMixin, {
       this.urlForRemoveFromHasMany(model, relationshipName),
       verb,
       { data: { data }}
-    )
+    ).then(() => {
+      if (relationshipKind === 'belongsTo') {
+        model.set(relationshipName, null)
+      } else {
+        model.get(relationshipName).removeObject(relatedModel)
+      }
+
+      if (inverseRelationshipKind === 'belongsTo') {
+        relatedModel.set(inverseRelationshipName, null)
+      } else if (inverseRelationshipKind) {
+        relatedModel.get(inverseRelationshipName).removeObject(relatedModel)
+      }
+    })
+  },
+  addRelated(model, relationshipName, relatedModel) {
+    const {
+      inverseRelationshipName,
+      inverseRelationshipKind,
+      relationshipKind
+    } = this._relationshipInfo(model, relationshipName)
+    let verb
+    let data = {
+      type: this._payloadKeyFromModel(relatedModel),
+      id: relatedModel.get('id')
+    }
+    if (relationshipKind === 'belongsTo') {
+      verb = 'PATCH'
+    } else {
+      verb = 'POST'
+      data = [data]
+    }
+
+    return this.ajax(this.urlForAddToHasMany(model, relationshipName), verb, {
+      data: { data }
+    }).then(() => {
```

Comment:
You think we should use the response objects sent back from the server?

like `.then(updatedModel => updatedModel.set...)`

Reply:
Jsonapi allows responses on relationship routes to return `204` no content. This approach works for any successful response format the server chooses.